kythe/rust/extractor/src/lib.rs

```diff
@@ -0,0 +1,101 @@
+// Copyright 2020 The Kythe Authors. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+#![feature(rustc_private)]
+
+extern crate rustc_driver;
+extern crate rustc_errors;
+extern crate rustc_interface;
+extern crate rustc_save_analysis;
+extern crate rustc_session;
+
+use rustc_driver::run_compiler;
+use rustc_driver::{Callbacks, Compilation};
+use rustc_interface::{interface, Queries};
+use rustc_save_analysis::DumpHandler;
+use std::path::PathBuf;
+
+/// Generate a save_analysis in `output_dir`
+///
+/// `rustc_arguments` is a `Vec<String>` containing Rust compiler arguments. The
+/// first element must be an empty string.
```

Comment:
If the first element must be an empty string, how about adding an explicit check and erroring out if it is not?